pkg/brokerpak/cmd.go

```diff
@@ -0,0 +1,195 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package brokerpak
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/tabwriter"
+
+	"github.com/GoogleCloudPlatform/gcp-service-broker/pkg/broker"
+	"github.com/GoogleCloudPlatform/gcp-service-broker/pkg/client"
+	"github.com/GoogleCloudPlatform/gcp-service-broker/pkg/generator"
+	"github.com/GoogleCloudPlatform/gcp-service-broker/pkg/providers/tf"
+	"github.com/GoogleCloudPlatform/gcp-service-broker/utils/stream"
+	"github.com/GoogleCloudPlatform/gcp-service-broker/utils/ziputil"
+	"github.com/spf13/viper"
+)
+
+// Init initializes a new brokerpak in the given directory with an example manifest and service definition.
+func Init(directory string) error {
+	exampleManifest := NewExampleManifest()
+	if err := stream.Copy(stream.FromYaml(exampleManifest), stream.ToFile(directory, manifestName)); err != nil {
+		return err
+	}
+
+	for _, path := range exampleManifest.ServiceDefinitions {
+		if err := stream.Copy(stream.FromYaml(tf.NewExampleTfServiceDefinition()), stream.ToFile(directory, path)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Pack creates a new brokerpak from the given directory which MUST contain a
+// manifest.yml file. If the pack was successful, the returned string will be
+// the path to the created brokerpak.
+func Pack(directory string) (string, error) {
+	abs, err := filepath.Abs(directory)
+	if err != nil {
+		return "", err
+	}
+	manifestPath := filepath.Join(directory, manifestName)
+	manifest, err := OpenManifest(manifestPath)
+	if err != nil {
+		return "", err
+	}
+
+	packname := filepath.Base(abs) + ".brokerpak"
+	return packname, manifest.Pack(directory, packname)
+}
+
+// Info writes out human-readable information about the brokerpak.
+func Info(pack string) error {
+	brokerPak, err := OpenBrokerPak(pack)
+	if err != nil {
+		return err
+	}
+
+	mf, err := brokerPak.Manifest()
+	if err != nil {
+		return err
+	}
+
+	services, err := brokerPak.Services()
+	if err != nil {
+		return err
+	}
+
+	// Pack information
+	fmt.Println("Information")
+	{
+		iw := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', tabwriter.StripEscape)
+		fmt.Fprintf(iw, "format\t%d\n", mf.PackVersion)
+		fmt.Fprintf(iw, "name\t%s\n", mf.Name)
+		fmt.Fprintf(iw, "version\t%s\n", mf.Version)
+		fmt.Fprintln(iw, "platforms")
+		for _, arch := range mf.Platforms {
+			fmt.Fprintf(iw, "\t%s\n", arch.String())
+		}
+		fmt.Fprintln(iw, "metadata")
+		for k, v := range mf.Metadata {
+			fmt.Fprintf(iw, "\t%s\t%s\n", k, v)
+		}
+
+		iw.Flush()
+		fmt.Println()
+	}
+
+	{
+		fmt.Println("Dependencies")
+		rw := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', tabwriter.StripEscape)
```

Comment:
This configuration of tabwriter seems to be repeated throughout. Consider making a factory which all of these locations all with this default config, in-case you want to change it later.